Microsoft.Toolkit.Uwp.UI.Controls/ImageCropper/ImageCropper.Helpers.cs

```diff
@@ -0,0 +1,261 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Graphics.Imaging;
+using Windows.Storage;
+using Windows.Storage.Streams;
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// The <see cref="ImageCropper"/> control allows user to crop image freely.
+    /// </summary>
+    public partial class ImageCropper
+    {
+        /// <summary>
+        /// Gets the cropped image.
+        /// </summary>
+        /// <param name="writeableBitmap">The source image.</param>
+        /// <param name="croppedRect">The cropped area.</param>
+        /// <returns>CroppedBitmap</returns>
+        internal static async Task<WriteableBitmap> GetCroppedBitmapAsync(
+            WriteableBitmap writeableBitmap,
+            Rect croppedRect)
+        {
+            var x = (uint)Math.Floor(croppedRect.X);
+            var y = (uint)Math.Floor(croppedRect.Y);
+            var width = (uint)Math.Floor(croppedRect.Width);
+            var height = (uint)Math.Floor(croppedRect.Height);
+            WriteableBitmap croppedBitmap;
+            var sourceStream = writeableBitmap.PixelBuffer.AsStream();
+            var buffer = new byte[sourceStream.Length];
+            await sourceStream.ReadAsync(buffer, 0, buffer.Length);
+            using (var memoryRandom = new InMemoryRandomAccessStream())
+            {
+                var encoder = await BitmapEncoder.CreateAsync(BitmapEncoder.PngEncoderId, memoryRandom);
+                encoder.SetPixelData(BitmapPixelFormat.Bgra8, BitmapAlphaMode.Premultiplied, (uint)writeableBitmap.PixelWidth, (uint)writeableBitmap.PixelHeight, 96.0, 96.0, buffer);
+                encoder.BitmapTransform.Bounds = new BitmapBounds
+                {
+                    X = x,
+                    Y = y,
+                    Height = height,
+                    Width = width
+                };
+                await encoder.FlushAsync();
+                croppedBitmap = new WriteableBitmap(
+                    (int)encoder.BitmapTransform.Bounds.Width,
+                    (int)encoder.BitmapTransform.Bounds.Height);
+                croppedBitmap.SetSource(memoryRandom);
+            }
+
+            return croppedBitmap;
+        }
+
+        /// <summary>
+        /// Gets the closest point in the rectangle to a given point.
+        /// </summary>
+        /// <param name="targetRect">The rectangle.</param>
+        /// <param name="point">The test point.</param>
+        /// <returns>A point within a rectangle.</returns>
+        internal static Point GetSafePoint(Rect targetRect, Point point)
```

Comment:
Why is this internal?

Reply:
Yes, that's wrong. It should be private.